### PR TITLE
fix(Semantic Release): Prevent overwrite of bundled dist/package.json [CICD-199]

### PR DESCRIPTION
**Pull request by christopher-learningpool**

### CODE REVIEW

**Review by brian-learningpool — APPROVED** [review]

File: scripts/public-circleci-lib-dist.js [hunk 1]
```diff
@@ -2,13 +2,36 @@
...
+}
+
 const main = async () => {
   try {
-    await Promise.all([
-      fs.copy('./package.json', './dist/package.json'),
-      fs.copy('./readme.md', './dist/readme.md'),
-      fs.copy('./license', './dist/license'),
-    ]);
+    await Promise.all(await getDistFiles());
```
Comment (ryasmi) [inline scripts/public-circleci-lib-dist.js]:
```suggestion
    await Promise.all([
      copyBestPath(['./package.json'], './dist/package.json')
      copyBestPath(['./readme.md', './README.md'], './dist/README.md'),
      copyBestPath(['./license', './LICENSE'], './dist/LICENSE'),
    ]);
```

File: scripts/public-circleci-lib-dist.js [hunk 2]
```diff
@@ -2,13 +2,36 @@
...
+    distfiles.push(fs.copy('./license', './dist/license'));
+  } else if (fs.existsSync('./LICENSE')) {
+    distfiles.push(fs.copy('./LICENSE', './dist/LICENSE'));
+  } else {
+    console.log('No license/LICENSE found.');
+  }
+
+  return distfiles;
+}
+
```
Comment (ryasmi) [inline scripts/public-circleci-lib-dist.js]:
```suggestion
async function copyBestPath(bestPaths, targetPath) {
  const bestPath = bestPaths.find((potentialBestPath) => {
    return fs.existsSync(potentialBestPath);
  });
  if (bestPath === undefined) {
    console.log(`Could not find a best path for ${targetPath} from ${bestPaths.join(' ')}`);
    return;
  }
  return fs.copy(bestPath, targetPath);
}
```

**Review by ryasmi — APPROVED** [review]
Nice @christopher-learningpool. LGTM, however, I've added some suggestions you can consider that might make it easier to extend in future and [avoid mutations](https://learninglocker.atlassian.net/wiki/spaces/LL/pages/1060503651/Immutable+Data).

File: scripts/public-circleci-lib-dist.js (hunk 2 above)
Comment (christopher-learningpool) [reply scripts/public-circleci-lib-dist.js]: Implemented feedback in the latest commit. Thank you again @ryansmith94

File: scripts/public-circleci-lib-dist.js [hunk 3]
```diff
@@ -2,36 +2,32 @@
...
-async function getDistFiles() {
-  const distfiles = [];
-
-  // Don't override the dist/package.json if it's already bundled.
-  if (!fs.existsSync('./dist/package.json')) {
-    distfiles.push(fs.copy('./package.json', './dist/package.json'));
+async function copyBestPath(bestPaths, targetPath, overwrite = true) {
+  if (!overwrite
+     && fs.existsSync(targetPath)) {
+    console.log(`Skipping ${bestPaths} as ${targetPath} has targeted file.`);
```
Comment (brian-learningpool) [inline scripts/public-circleci-lib-dist.js]: Should there be `${bestPaths.join(' ')}` here?

Comment (christopher-learningpool) [reply scripts/public-circleci-lib-dist.js]: Good spot!

**Pull request merged**